src/Service/PdfCreator.php

```diff
@@ -4,38 +4,39 @@
 
 namespace App\Service;
 
-use Spipu\Html2Pdf\Html2Pdf;
+use Dompdf\Dompdf;
+use Dompdf\Options;
 
 final class PdfCreator implements PdfCreatorInterface
 {
     private $pdf;
     private $orientation;
     private $format;
-    private $lang;
-    private $unicode;
-    private $encoding;
-    private $margin;
 
-    public function __construct(string $orientation, string $format, string $lang, bool $unicode, string $encoding, array $margin)
+    public function __construct(string $orientation, string $format)
     {
         $this->orientation = $orientation;
         $this->format = $format;
-        $this->lang = $lang;
-        $this->unicode = $unicode;
-        $this->encoding = $encoding;
-        $this->margin = $margin;
     }
 
-    public function create(): void
+    public function create(bool $isHtml5ParserEnabled = true, bool $isRemoteEnabled = true, Options $options = null): void
     {
-        $this->pdf = new Html2Pdf($this->orientation, $this->format, $this->lang, $this->unicode, $this->encoding, $this->margin);
-        $this->pdf->setTestIsImage(false);
+        if (null === $options) {
+            $options = new Options([
+                'isHtml5ParserEnabled' => $isHtml5ParserEnabled,
+                'isRemoteEnabled' => $isRemoteEnabled,
+            ]);
+        }
+        $this->pdf = new Dompdf($options);
+        $this->pdf->setPaper($this->format, $this->orientation);
     }
 
-    public function generatePdf(string $template, string $name): string
+    public function generatePdf(string $template, string $name): void
     {
-        $this->pdf->writeHTML($template);
-
-        return $this->pdf->Output($name.'.pdf');
+        $this->pdf->loadHtml($template);
+        $this->pdf->render();
+        $this->pdf->stream($name.'.pdf', [
```

Comment:
Do not stream it to the user, you're in a service, you should not expect to output for a user. It could be used to store the file somewhere afterward.

tl;dr: use `output` rather than `stream`, and return the string.